kratos/CMakeLists.txt

```diff
@@ -2,188 +2,39 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/read_materials_utility.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/variable_redistribution_utility.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/auxiliar_model_part_utilities.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/time_discretization.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/constraint_utilities.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/sensitivity_builder.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/mortar_utilities.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/exact_mortar_segmentation_utility.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/timer.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/geometrical_transformation_utilities.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/delaunator_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/properties_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/string_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/result_dabatase.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/geometry_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/normal_calculation_utils.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/modeler/tetrahedra_ball.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modeler/tetrahedra_edge_shell.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modeler/cad_io_modeler.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modeler/modeler_factory.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/ausas_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/triangle_2d_3_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/triangle_2d_3_ausas_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/tetrahedra_3d_4_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/tetrahedra_3d_4_ausas_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/elements/embedded_nodal_variable_calculation_element_simplex.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/elements/mesh_element.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/conditions/mesh_condition.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/processes/apply_periodic_boundary_condition_process.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/linear_solvers/amgcl_solver_impl.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/communication_coloring_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/constraints/slip_constraint.cpp
+SET(LIST_OF_SOURCE_EXCEPTIONS
+    ${CMAKE_CURRENT_SOURCE_DIR}/processes/assign_scalar_field_to_entities_process.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/sources/kratos_version.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/processes/mesh_smoothing_process.cpp # This wasn't in the original list for some reason
+    ${CMAKE_CURRENT_SOURCE_DIR}/sources/kratos_exception.cpp # This wasn't in the original list for some reason
```

Comment:
why do you leave them separate?

Reply:
`sources/kratos_version.cpp` is added manually later (I don't know the reason, @roigcarlo do you know?). `processes/assign_scalar_field_to_entities_process.cpp` is beacsue depends on python.h  (that's why is added to python sources). `processes/mesh_smoothing_process.cpp` was not added in the original one, and in fact if you compile with cotire it crashes (I don't know why, but as I said is not currently in the list). `sources/kratos_exception.cpp` wasn't in the original list (I don't know why, but if i add it at least it compiles)

Reply:
**kratos_version** is excluded because I separated it to another CU so you didn't have to recompile the whole core every time the version changed, just relink it. In order to not pollute with more `.dll / .so` a separate `.o` is created and then using during the linking of `KratosCore.dll`

**kratos_exception** ... wasn't this disabled? I remember that we had problems while compiling with this in visual studio, so we left it commented. The reason was because that particular feature was incompatible with some other flags, but can't remember which ones exactly.

Reply:
I mean, its ok  for those to be excluded, they should not appear in the list of sources to be compiled. (Reading my last message seems that I am implying that kratos_exception is used)

Reply:
So, do you agree?

Reply:
I added your explanation as a documentation

Reply:
> So, do you agree?

I mean we can proceed with this before conflicts arise